### ISSUE TITLE
[t] 'true' value must be either True or False." in network.py 

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2366]
N.B. [t] = testing channel. Post the proposed Django update in pending #2365 we have uncovered a prior silent 'true' (string type) to True (Boolean) silent conversion that the proposed Django update is now catching. In this case due to our so far adequate extraction of the icc field in the following:
https://github.com/rockstor/rockstor-core/blob/eeb0084a2ea6f6bdb1cdbe35428aa5021ddadc72/src/rockstor/system/network.py#L234-L237
We look to have the same, yet to be surfaced issue regarding the ip_masquerade field in the same area:
https://github.com/rockstor/rockstor-core/blob/eeb0084a2ea6f6bdb1cdbe35428aa5021ddadc72/src/rockstor/system/network.py#L239-L244

It is proposed that we further pre-process these fields to render them boolean to match our associated models. We should also note in the code that any future use of the default_bridge field/element also be treated likewise given it's similar string nature but boolean meaning.
> u'com.docker.network.bridge.default_bridge': u'true'

The fields and their type are set likely due to upstream constrains or bugs via the command line wrapper used in the following:

https://github.com/rockstor/rockstor-core/blob/81a0d5f5e0a6dfe5a922199828f66eeab0253e65/src/rockstor/system/docker.py#L112-L121

Where DNET is the "docker network" system command.

Thanks to @FroggyFlox for their assistance in diagnosing and confirming this pending issue post indicated pr merge.
See the discussion in the above referenced and currently on-going pull request starting from:
https://github.com/rockstor/rockstor-core/pull/2365#issuecomment-1093158200

## Reproducer:
On an existing rockstor install with at least one Rock-on installed, in order to have the associated docker network setup in place. Do a complete install reset via:
```
systemctl stop rockstor rockstor-pre rockstor-bootstrap
userdel prior-admin-user-name-here
rm /opt/rockstor-dev/.initrock
systemctl start rockstor
```

Post referenced pull request with Django update and initial setup screen admin user create we receive the following:

> Houston, we've had a problem.
> [u"'true' value must be either True or False."]

with the associated traceback:
```
[08/Apr/2022 19:02:42] DEBUG [storageadmin.views.network:82] The function _update_or_create_ctype has been called
[08/Apr/2022 19:02:42] ERROR [storageadmin.util:45] Exception: [u"'true' value must be either True or False."]
Traceback (most recent call last):
  File "/opt/rockstor-dev/src/rockstor/rest_framework_custom/generic_view.py", line 41, in _handle_exception
    yield
  File "/opt/rockstor-dev/src/rockstor/storageadmin/views/network.py", line 600, in post
    self._refresh_connections()
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/utils/decorators.py", line 185, in inner
    return func(*args, **kwargs)
  File "/opt/rockstor-dev/src/rockstor/storageadmin/views/network.py", line 184, in _refresh_connections
    cls._update_or_create_ctype(nco, ctype, ctype_d)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/utils/decorators.py", line 185, in inner
    return func(*args, **kwargs)
  File "/opt/rockstor-dev/src/rockstor/storageadmin/views/network.py", line 129, in _update_or_create_ctype
    connection=co, usercon=usercon, **config
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/query.py", line 394, in create
    obj.save(force_insert=True, using=self.db)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/base.py", line 808, in save
    force_update=force_update, update_fields=update_fields)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/base.py", line 838, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/base.py", line 924, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/base.py", line 963, in _do_insert
    using=using, raw=raw)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/query.py", line 1079, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/sql/compiler.py", line 1111, in execute_sql
    for sql, params in self.as_sql():
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/sql/compiler.py", line 1064, in as_sql
    for obj in self.query.objs
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/sql/compiler.py", line 1003, in prepare_value
    value = field.get_db_prep_save(value, connection=self.connection)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/fields/__init__.py", line 770, in get_db_prep_save
    prepared=False)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/fields/__init__.py", line 762, in get_db_prep_value
    value = self.get_prep_value(value)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/fields/__init__.py", line 1043, in get_prep_value
    return self.to_python(value)
  File "/opt/rockstor-dev/eggs/Django-1.11.29-py2.7.egg/django/db/models/fields/__init__.py", line 1036, in to_python
    params={'value': value},
ValidationError: [u"'true' value must be either True or False."]
[08/Apr/2022 19:02:42] DEBUG [storageadmin.util:46] Current Rockstor version: Unknown Version
```
With some additional confirmation via debug logging added to further investigate:
```
[08/Apr/2022 20:35:54] DEBUG [system.network:238] dtmap['Options']={u'com.docker.network.bridge.name': u'docker0', u'com.docker.network.bridge.default_bridge': u'true', u'com.docker.network.bridge.enable_ip_masquerade': u'true', u'com.docker.network.driver.mtu': u'1500', u'com.docker.network.bridge.host_binding_ipv4': u'0.0.0.0', u'com.docker.network.bridge.enable_icc': u'true'}
[08/Apr/2022 20:35:54] DEBUG [storageadmin.views.network:176] 'ctype' was found in config
[08/Apr/2022 20:35:54] DEBUG [storageadmin.views.network:179] we have extracted a ctype_d={'aux_address': None, 'subnet': u'172.17.0.0/16', 'internal': False, 'host_binding': u'0.0.0.0', 'docker_name': 'docker0', 'icc': u'true', 'ip_masquerade': False, 'dgateway': u'172.17.0.1', 'ip_range': None}
[08/Apr/2022 20:35:54] DEBUG [storageadmin.views.network:82] The function _update_or_create_ctype has been called
[08/Apr/2022 20:35:54] DEBUG [storageadmin.views.network:83] co=NetworkConnection object, ctype=bridge, config={'aux_address': None, 'subnet': u'172.17.0.0/16', 'internal': False, 'host_binding': u'0.0.0.0', 'docker_name': 'docker0', 'icc': u'true', 'ip_masquerade': False, 'dgateway': u'172.17.0.1', 'ip_range': None}
[08/Apr/2022 20:35:54] ERROR [storageadmin.util:45] Exception: [u"'true' value must be either True or False."]
```

**Comment by phillxnet** [comment]
After applying a proposed fix for his issue we have roughly equivalent temporary debug output as follows:
```
[09/Apr/2022 16:13:23] DEBUG [system.network:240] dtmap['Options']={u'com.docker.network.bridge.name': u'docker0', u'com.docker.network.bridge.default_bridge': u'true', u'com.docker.network.bridge.enable_ip_masquerade': u'true', u'com.docker.network.driver.mtu': u'1500', u'com.docker.network.bridge.host_binding_ipv4': u'0.0.0.0', u'com.docker.network.bridge.enable_icc': u'true'}
[09/Apr/2022 16:13:23] DEBUG [system.network:243] tmap[tmap["ctype"]]["icc"]=True
[09/Apr/2022 16:13:23] DEBUG [storageadmin.views.network:176] 'ctype' was found in config
[09/Apr/2022 16:13:23] DEBUG [storageadmin.views.network:179] we have extracted a ctype_d={'aux_address': None, 'subnet': u'172.17.0.0/16', 'internal': False, 'host_binding': u'0.0.0.0', 'docker_name': 'docker0', 'icc': True, 'ip_masquerade': False, 'dgateway': u'172.17.0.1', 'ip_range': None}
[09/Apr/2022 16:13:23] DEBUG [storageadmin.views.network:82] The function _update_or_create_ctype has been called
[09/Apr/2022 16:13:23] DEBUG [storageadmin.views.network:83] co=NetworkConnection object, ctype=bridge, config={'aux_address': None, 'subnet': u'172.17.0.0/16', 'internal': False, 'host_binding': u'0.0.0.0', 'docker_name': 'docker0', 'icc': True, 'ip_masquerade': False, 'dgateway': u'172.17.0.1', 'ip_range': None}
```
So we have a successful conversion of our string to bool type for both the "icc" and "ip_masquerade" fields.
Notice that the ip_masquerade value is still inverted. This is functionally the same as before so should not cause an issue.

**Comment by phillxnet** [comment]
As the proposed fix for this issue is on top of the indicated large pending Django update pr I will submit a pull request after having merged that dependant code.

**Comment by phillxnet** [comment]
Closing as fixed by #2367
